kubeIndicator.js

```diff
@@ -5,67 +5,83 @@ import Clutter from 'gi://Clutter';
 import GLib from 'gi://GLib';
 import Gio from 'gi://Gio';
 import GObject from 'gi://GObject';
+import { gettext as _ } from 'resource:///org/gnome/shell/extensions/extension.js';
 
 import { KubePopupMenuItem } from './kubePopupMenuItem.js';
-import { Yaml } from './lib/yaml/Yaml.js';
+import { KubectlConfig } from './kubectl.js';
+
 
 export const KubeIndicator = GObject.registerClass({ GTypeName: 'KubeIndicator' },
     class KubeIndicator extends PanelMenu.Button {
         _init(extensionObject) {
             super._init(null, "Kube");
             this._extensionObject = extensionObject
             this._settings = this._extensionObject.getSettings();
-            this.kcPath = GLib.get_home_dir() + "/.kube/config";
+
+            this._monitors = [];
 
             this._setView()
 
-            let kcFile = Gio.File.new_for_path(this.kcPath);
-            this._monitor = kcFile.monitor(Gio.FileMonitorFlags.NONE, null);
-            this._monitor.connect('changed', this._update.bind(this));
+            let kConfigFiles = [];
+
+            if (GLib.getenv('KUBECONFIG') !== null) {
```

Comment:
Did you test this? Is the variable available for the extension in the context of gnome shell if it is set in `.bashrc` or `.zshrc`?

Reply:
I don't remember where I originally read why it's a good practice to put user environment variables in `$HOME/.profile`, but you can find explanation around the web, e.g. this [post](https://askubuntu.com/a/1411917).

BTW, yes, I test `KUBECONFIG` env var in `$HOME/.profile` and it's accessible from within gnome-shell extension.